Arm Subsystem

YehudaRothstein opened the pull request:
This pull request introduces new classes and constants for the arm subsystem of the robot. The most significant changes include the addition of the `Constants` class to define various hardware IDs and control parameters, and the implementation of the `Placer` class to manage the arm's motors and mechanisms.

New additions to the arm subsystem:

* [`src/main/java/frc/robot/subsystems/arm/Constants.java`](diffhunk://#diff-dfd7d32536130450031de2202c70de6c6cddddeb8e726e6e7533a68f3d16aaadR1-R34): Added a new `Constants` class to define hardware IDs, control gains, and other constants for the arm subsystem.

Implementation of the `Placer` class:

* [`src/main/java/frc/robot/subsystems/arm/Placer.java`](diffhunk://#diff-4b7a4298f98e7b760eafa97ca5c24bd449c183a8b8813cdee995591a57f4dc74R1-R49): Introduced the `Placer` class, which extends `SubsystemBase` and manages the arm's motors, encoder, and mechanisms using the constants defined in the `Constants` class.Co-authored-by: Eliya1452 <eliyabl2008@gmail.com>

YehudaRothstein committed:
add: Placer subsystem
Co-authored-by: Eliya1452 <eliyabl2008@gmail.com>

Keller6738 commented on src/main/java/frc/robot/subsystems/arm/Placer.java:
```diff
@@ -0,0 +1,49 @@
+        m_angleEncoder = new CANcoder(ANGLE_CANCODER_ID);
+        m_radSupplier = ()-> m_angleEncoder.getPosition().getValueAsDouble()*2 * PI;
+        m_arm = new Arm(
+                new MotorGroup(m_firstRotationMotor, m_secondRotationMotor),
```
create a field of this motor group

Keller6738 commented on src/main/java/frc/robot/subsystems/arm/Placer.java:
```diff
@@ -0,0 +1,49 @@
+        m_radSupplier = ()-> m_angleEncoder.getPosition().getValueAsDouble()*2 * PI;
+        m_arm = new Arm(
+                new MotorGroup(m_firstRotationMotor, m_secondRotationMotor),
+                ()-> m_angleEncoder.getPosition().getValueAsDouble(),
```
call the `m_radSupplier` instead

eliya1452 replied:
its in degrees so instead of calling m_radSupplier I opted for making a new variable that is a m_angleSupplier

Keller6738 replied:
If this is in degrees, change to radians
We are working ONLY with radians.

eliya1452 replied:
My bad, the function incorrectly received in the way Yehuda did it an angle instead of radians so i thought it should receive an angle but i was wrong; it should receive radians, so i changed it and removed the new variable per your request.

eliya1452 committed:
Refactor Placer class: Simplify motor initialization and remove redundant code

- Moved motor initializations for `m_firstRotationMotor` and `m_secondRotationMotor` outside of the constructor to avoid error in motorGroup initialization.
- Changed the method of retrieving the encoder position by using `m_angleEncoder.getPosition().getValueAsDouble()` directly in the `Arm` constructor and `m_radSupplier` and fixing the problem of it receiving degrees instead of radians.
- Added ManualCommand to Placer; and also added goToAngle without Tolerance Consumer.

eliya1452 committed:
Update Placer.java

fix:
ManualCommand without making a new command variable but returning it
goToAngle name change and made it receive toleranceConsumer
m_firstRotationMotor, m_secondRotationMotor and m_motorGroup be initialized inside the constructor
Co-authored-by: Keller6738 <itaykeller12345@gmail.com>

eliya1452 committed:
Update Placer.java

feature:
Consumer<Boolean> toleranceConsumer to the goToAngleCommand + added MAX_OFFSET constant

eliya1452 committed:
Update Constants.java

added MAX_OFFSET

eliya1452 committed:
Update Arm.java

added maxOffSet variable instead of being recieved in the function call bluntly in code.

Keller6738 committed:
fix: added gripperWheels Voltage to manualCommand

Keller6738 committed:
add: removeAlgae Command

Keller6738 committed:
add: inputCoral Command

Keller6738 committed:
add: outputCoral Command

Keller6738 committed:
add: Constants for input/output Coral Commands

YehudaRothstein committed:
fix: changed Remove algae output

YehudaRothstein committed:
fix: changed Remove algae output in Constants

ExcaliburFRC6738 committed:
fix: added limit to angle Supplier

ExcaliburFRC6738 committed:
update: formated Constants
